scripts/admin/launch-meteor

```diff
@@ -82,15 +82,36 @@ if [ ! -x "$METEOR_WAREHOUSE_DIR/meteor" ]; then
   fi
   echo "Installing a Meteor distribution in your home directory." 1>&2
 
-  # Only show progress bar animations if we have a tty
-  # (Prevents tons of console junk when installing within a pipe)
-  if [[ -t 1 ]]; then
-    curl --progress-bar --fail "$TARBALL_URL" | tar -xzf - -C "$INSTALL_TMPDIR"
-  else
-    curl -s --fail "$TARBALL_URL" | tar -xzf - -C "$INSTALL_TMPDIR"
-  fi
+  TARBALL_FILE="$(dirname "$METEOR_WAREHOUSE_DIR")/.meteor-tarball-tmp"
+  rm -rf "$TARBALL_FILE"
+
+  echo "Downloading Meteor distribution"
+  # keep trying to curl the file until it works (resuming where possible)
+  MAX_ATTEMPTS=99
+  set +e
+  CODE=1
+  ATTEMPTS=0
+  while [ $CODE -ne 0 ] && [ $ATTEMPTS -lt $MAX_ATTEMPTS ]
+  do
+    ATTEMPTS=$((ATTEMPTS + 1))
+
+    # Only show progress bar animations if we have a tty
+    # (Prevents tons of console junk when installing within a pipe)
+    if [ -t 1 ]; then
+      curl --progress-bar --fail "$TARBALL_URL" -C - -o "$TARBALL_FILE"
+    else
+      curl -s --fail "$TARBALL_URL" -C - -o "$TARBALL_FILE"
+    fi
+
+    CODE=$?
+  done
+  set -e
 
   # bomb out if it didn't work, eg no net
+  test -e "${TARBALL_FILE}"
+  tar -xzf "$TARBALL_FILE" -C "$INSTALL_TMPDIR" -o
+  rm -rf "$TARBALL_FILE"
```

Comment:
We should `rm -rf "$TARBALL_FILE"` even if (especially if) the extraction failed.